hazelcast/src/main/java/com/hazelcast/nio/tcp/nonblocking/AbstractHandler.java

```diff
@@ -56,14 +59,15 @@ public AbstractHandler(TcpIpConnection connection, NonBlockingIOThread ioThread,
         this.connection = connection;
         this.ioThread = ioThread;
         this.ioThreadId = ioThread.id;
-        this.selector = ioThread.getSelector();
         this.socketChannel = connection.getSocketChannelWrapper();
         this.connectionManager = connection.getConnectionManager();
         this.ioService = connectionManager.getIoService();
         this.logger = ioService.getLogger(this.getClass().getName());
         this.initialOps = initialOps;
     }
 
+    protected abstract void reschedule();
```

Comment:
Probably this method doesn't need to be implemented at every subclass. We should be able to implement it directly.

I need to think about the implementation, because it could bite with the migration.